crates/viewer/re_viewport/src/system_execution.rs

```diff
@@ -124,22 +119,31 @@ pub fn execute_systems_for_all_views<'a>(
     ctx: &'a ViewerContext<'a>,
     tree: &egui_tiles::Tree<SpaceViewId>,
     views: &'a BTreeMap<SpaceViewId, SpaceViewBlueprint>,
-    view_states: &ViewStates,
+    view_states: &mut ViewStates,
 ) -> HashMap<SpaceViewId, (ViewQuery<'a>, SystemExecutionOutput)> {
     let Some(time_int) = ctx.rec_cfg.time_ctrl.read().time_int() else {
         return Default::default();
     };
 
     re_tracing::profile_wait!("execute_systems");
 
+    // During system execution we only have read access to the view states, so we need to ensure they exist ahead of time.
+    for (view_id, view) in views {
+        view_states.ensure_state_exists(*view_id, view.class(ctx.space_view_class_registry));
+    }
+
     tree.active_tiles()
         .into_par_iter()
         .filter_map(|tile_id| {
             tree.tiles.get(tile_id).and_then(|tile| match tile {
-                egui_tiles::Tile::Pane(view_id) => views.get(view_id).map(|view| {
-                    let result = execute_systems_for_space_view(ctx, view, time_int, view_states);
-
-                    (*view_id, result)
+                egui_tiles::Tile::Pane(view_id) => views.get(view_id).and_then(|view| {
+                    let Some(view_state) = view_states.get(*view_id) else {
+                        debug_assert!(false, "View state for view {view_id:?} not found. That shouldn't be possible since we just ensured they exist above.");
```

Comment:
some rusty way to avoid this debug assert would be nice, but can't think of one right now.
Also above loop iterates over _all_ views which is technically more than needed - we don't have to do that for any that aren't visible